Change the response format from logger output to a structured JSON response

TysonAndre opened the pull request:
By using a JSON response format, this allows uniqush clients to know if the API was successful.
API clients will be able to easily determine the reason an API failed, without resorting to heuristics (checking if the response contains a certain string).
Additionally, it is much easier for clients to process the outcome of a push, when a push involves more than one device.

- bump version from 1.5.x to 1.6.0, since the API response format changed. No change in functionality or api request format. **(or should this be a major version change, instead?)**

(This is mostly the same as what we use internally. 
The original author's commits were edited to remove a field that wasn't used by upstream. 
Additionally, `go fmt` was run on every commit, and some commits were squashed)

Barely related to issue #55 (Related in the sense that the responses are much easier to parse)

Also fixes #71 

### Examples

Example return value of `/subscribe` (after indenting), with fake data

```json
{
  "errorMsg": null,
  "code": "UNIQUSH_SUCCESS",
  "requestId": null,
  "service": "fooserv",
  "from": "a.b.c.d:39317",
  "subscriber": "foo.1234",
  "pushServiceProvider": "apns:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
  "deliveryPoint": "apns:bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb",
  "messageId": null
}
```

Example return value of `/push`

```json
{
  "droppedDetails": [],
  "failureDetails": [],
  "successDetails": [
    {
      "code": "UNIQUSH_SUCCESS",
      "requestId": "56c3cb5f-RnIk4aFK4c7xKjcwtt143Q==",
      "service": "fooserv",
      "from": "a.b.c.d:37360",
      "subscriber": "foo.1234",
      "pushServiceProvider": "apns:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
      "deliveryPoint": "apns:bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb",
      "messageId": "apns:apns:cccccccccccccccccccccccccccccccccccccccc-78"
    }
  ],
  "droppedCount": 0,
  "failureCount": 0,
  "successCount": 1,
  "date": 1455672159,
  "type": "Push"
}
```

Example return value of `/addpsp`:

```json
{
  "details": {
    "code": "UNIQUSH_SUCCESS",
    "messageId": null,
    "deliveryPoint": null,
    "pushServiceProvider": "apns:8fa8483afc5f4461117c8c2d61b36e7b6c8d3eb2",
    "subscriber": null,
    "from": "127.0.0.1:42532",
    "service": "myservice",
    "requestId": null
  },
  "status": 0,
  "date": 1455672625,
  "type": "AddPushServiceProvider"
}
```

EDIT: the comment about including errorMsg was outdated - this wasn't part of the original commit. It was added later, as part of a larger change.

mishan commented on main.go:
```diff
@@ -22,18 +22,19 @@ import (
 var uniqushPushShowVersionFlag = flag.Bool("version", false, "Version info")
 
-var uniqushPushVersion = "uniqush-push 1.5.2"
+var uniqushPushVersion = "uniqush-push 1.6.0"
```
Since it's a major change that could break old clients expecting the old style of reply, I'd personally opt for going up a major version to 2.0.0

mishan commented:
Code lgtm otherwise, I mean we have reviewed this extensively and are running it on production :) I'd say let's bump the version to 2.0.0. Could you perhaps rebase+squash David Jeong's commits, too?

TysonAndre commented:
Squashed David Jeong's commits.
Updated the version to 2.0.0.

Printed newline after the curl response in apns-test.sh(The response itself no longer ends with a newline)
